diag_manager/diag_manager.F90

```diff
@@ -4220,7 +4232,11 @@ SUBROUTINE diag_field_add_attribute_scalar_i(diag_field_id, att_name, att_value)
     CHARACTER(len=*), INTENT(in) :: att_name !< new attribute name
     INTEGER, INTENT(in) :: att_value !< new attribute value
 
-    CALL diag_field_add_attribute_i1d(diag_field_id, att_name, (/ att_value /))
+    if (use_modern_diag) then
+      call fms_diag_field_add_attribute(diag_field_id, att_name, (/att_value /))
```

Comment:
Is this a wrapper on an object procedure?  Can you just call the object procedure directly?  there are a few of these...

Reply:
fms_diag_field_add_attribute is a wrapper basically to avoid having to duplicate this code for each case:

https://github.com/NOAA-GFDL/FMS/blob/09a396e4cc62f19a7db9a9d1521513f13eb5302e/diag_manager/fms_diag_object.F90#L1143-L1149